app/models/pawn.rb

```diff
@@ -1,7 +1,10 @@
 class Pawn < Piece
   def valid_move?(x, y)
     is_obstructed?(x, y)
-    one_forward(x, y) || two_forward(x, y) || one_diagonal(x, y)
+    destination = game.pieces.where(x_position: x, y_position: y).take
+    (one_forward(x, y) && destination.nil?) ||
+    (two_forward(x, y) && destination.nil?) ||
+    (one_diagonal(x, y) && !destination.nil? && destination.color == !color)
   end
```

Comment:
A bit of refactoring in the Pawns model: now it should work as expected when a pawn tries to move on a friendly piece.
I edited the Pawns model/tests because were interfering with my Game model tests.